Rename $env.PATH to $env.Path to reflect actual shell behavior

debemdeboas opened the pull request:
The documentation states that `$env.PATH` lets the user quote "See the current path". This is not what happens though, the command errors out.

```
❯ $env.PATH
Error: nu::shell::name_not_found (link)

  × Name not found
   ╭─[entry #16:1:1]
 1 │ $env.PATH
   ·      ──┬─
   ·        ╰── did you mean 'Path'?
   ╰────
```

<details><summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/35022953/221650175-f6efc961-91e2-4664-b6f2-9231ff45e55a.png)

</details>

The correct command *in Windows* to get the current path is `$env.Path`, not `$env.PATH`.

fdncred commented:
This is a Windows vs *nix thing. On Windows, path is usually `Path`. On MacOS and Linux it's usually `PATH`.

debemdeboas commented:
You're right. Maybe adding both examples would be beneficial?

debemdeboas commented:
I changed it to be like so:

![image](https://user-images.githubusercontent.com/35022953/221659352-a1ba8759-50da-4bc8-82cd-1c9dbc35dab5.png)

fdncred commented:
Looks good but I wouldn't say posix since nushell isn't posix compatible. I'd just say MacOS and Linux.

fdncred commented:
You could maybe say `Non-Windows` - I'm not sure what the android path looks like but we have people who use it on android.

debemdeboas commented:
As soon as I get home I will change it to Non-Windows for us to see.

debemdeboas commented:
Sorry for the delay.
I changed it to (Non-Windows) and removed (Windows) since it was redundant.

![image](https://user-images.githubusercontent.com/35022953/222569820-7ced1993-e095-4f6e-b28e-43661b2a7395.png)

fdncred commented:
ok, i don't see the changes in the PR yet.

debemdeboas commented:
I didn't change it in the PR, I was waiting for your opinion :D

fdncred commented:
Thanks for helping out. I think it looks good enough for now.
